### DISCUSSION TITLE
I constantly have to ignore `.vim/*`, because I don't want an empty `coc-settings.json` in my repo ...

### DISCUSSION THREAD

**Discussion by martin-braun** [neoclide/coc.nvim#4636] [Q&A]
Hi,

I love coc.nvim, but for some reason it likes to create a `.vim/coc-settings.json` with `{}` as its contents, when I do not want to add settings anyways.

I use a [gitignore generator](https://gitignore.io), so it bothers me a little bit that I have to add `.vim/*` to avoid an empty `coc-settings.json` to be committed.

I'd rather have `coc.nvim` not to create a `coc-settings.json` that is empty, because once I have reasonable local settings in `coc-settings.json` I do think it should be committed.

How can I tell `coc.nvim` to not create empty `coc-settings.json`? Thanks.

**Comment by fannheyward** [comment]
coc.nvim didn't create `.vim/coc-settings.json` by default, it's created with `CocLocalConfig`.

**Reply by martin-braun** [reply]
I didn't call it. Could a plugin be responsible for that?

EDIT: I did a recursive grep on `$HOME/.config/coc/extensions/node_modules`. No `CocLocalConfig` anywhere. How can I figure out whats responsible for the creation of `coc-settings.json`?

**Reply by fannheyward** [reply]
Try minimal vimrc, extension won't call `CocLocalConfig` directly, maybe `openLocalConfig`.

What's your `:CocList extensions`?

**Reply by martin-braun** [reply]
`:CocList extensions`:

```
* coc-tailwindcss 0.7.0 ~/.config/coc/extensions/node_modules/coc-tailwindcss
* coc-snippets 3.1.6 ~/.config/coc/extensions/node_modules/coc-snippets
* coc-prettier 9.3.1 ~/.config/coc/extensions/node_modules/coc-prettier
* coc-pairs 1.4.2 ~/.config/coc/extensions/node_modules/coc-pairs
* coc-html 1.8.0 ~/.config/coc/extensions/node_modules/coc-html
* coc-highlight 2.0.4 ~/.config/coc/extensions/node_modules/coc-highlight
* coc-eslint 1.6.0 ~/.config/coc/extensions/node_modules/coc-eslint
* coc-emmet 1.1.6 ~/.config/coc/extensions/node_modules/coc-emmet
+ coc-yaml 1.9.0 ~/.config/coc/extensions/node_modules/coc-yaml
+ coc-xml 1.14.1 ~/.config/coc/extensions/node_modules/coc-xml
+ coc-vimlsp 0.13.0 ~/.config/coc/extensions/node_modules/coc-vimlsp
+ coc-tsserver 2.1.3 ~/.config/coc/extensions/node_modules/coc-tsserver
+ coc-svg 0.0.13 ~/.config/coc/extensions/node_modules/coc-svg
+ coc-sh 0.7.0 ~/.config/coc/extensions/node_modules/coc-sh
+ coc-rust-analyzer 0.72.0 ~/.config/coc/extensions/node_modules/coc-rust-analyzer
+ coc-pyright 1.1.291 ~/.config/coc/extensions/node_modules/coc-pyright
+ coc-phpls 2.2.2 ~/.config/coc/extensions/node_modules/coc-phpls
+ coc-lua 1.0.1 ~/.config/coc/extensions/node_modules/coc-lua
+ coc-json 1.8.0 ~/.config/coc/extensions/node_modules/coc-json
+ coc-jedi 0.35.0 ~/.config/coc/extensions/node_modules/coc-jedi
+ coc-java 1.6.0 ~/.config/coc/extensions/node_modules/coc-java
+ coc-flutter 1.9.9 ~/.config/coc/extensions/node_modules/coc-flutter
+ coc-deno 3.11.0 ~/.config/coc/extensions/node_modules/coc-deno
+ coc-css 2.0.0 ~/.config/coc/extensions/node_modules/coc-css
+ coc-blade 0.16.0 ~/.config/coc/extensions/node_modules/coc-blade
```

I use Neovim + https://github.com/NvChad/NvChad and here are my custom configs on top: https://github.com/martin-braun/CocChad. Right now I can't say the exact moment that the file gets created. I usually only realize this when I try to commit, I will try to make it reproducable with my configs later, but I can tell that I noticed this when working on https://github.com/martin-braun/google_maps_place_picker_mb, I'm not sure if its really due to `coc-flutter`, though.

**Reply by fannheyward** [reply]
coc-flutter has some _update_ configurations operations, this will save to `coc-settings.json`, but as you said, your `.vim/coc-settings.json` is empty.

@iamcco sorry to boring you, did coc-flutter will auto-update `.vim/coc-settings.json`?
